plugins/modules/swim_workflow_manager.py

```diff
@@ -1064,119 +1255,158 @@ def get_diff_import(self):
+                    if task_details and "completed successfully" in task_details.get("progress", "").lower():
+                        if images_to_import:
+                            images_to_import_str = ", ".join(images_to_import)
+
+                            self.result['changed'] = True
+                            self.status = "success"
+                            self.msg = "Swim Image(s) {0} imported successfully".format(images_to_import_str)
+                            self.result['msg'] = self.msg
+                            self.log(self.msg, "INFO")
+                            break
+
+                    if task_details and task_details.get("isError"):
+                        if "already exists" in task_details.get("failureReason", ""):
+                            self.msg = "SWIM Image {0} already exists in the Cisco Catalyst Center".format(image_name.split('/')[-1])
+                            self.result['msg'] = self.msg
+                            self.log(self.msg, "INFO")
+                            self.status = "success"
+                            self.result['changed'] = False
+                            break
+                        else:
+                            self.status = "failed"
+                            self.msg = task_details.get("failureReason", "SWIM Image {0} seems to be invalid".format(image_name))
+                            self.log(self.msg, "WARNING")
+                            self.result['response'] = self.msg
+                            return self
+
+                self.result['response'] = task_details if task_details else response
+
+                image_name = image_name.split('/')[-1]
+                image_id = self.get_image_id(image_name)
+                self.have["imported_image_id"] = image_id
 
-                if task_details and task_details.get("isError"):
-                    if "already exists" in task_details.get("failureReason", ""):
-                        self.msg = "SWIM Image {0} already exists in the Cisco Catalyst Center".format(name)
-                        self.result['msg'] = self.msg
-                        self.log(self.msg, "INFO")
-                        self.status = "success"
-                        self.result['changed'] = False
-                        break
-                    else:
-                        self.status = "failed"
-                        self.msg = task_details.get("failureReason", "SWIM Image {0} seems to be invalid".format(image_name))
-                        self.log(self.msg, "WARNING")
-                        self.result['response'] = self.msg
-                        return self
+            imported_images_str = ", ".join(images_to_import)
+            skipped_images_str = ", ".join(existing_images)
 
-            self.result['response'] = task_details if task_details else response
+            if skipped_images_str and not imported_images_str:
```

Comment:
we are checking skipped_images_str twice. Can we optimize it?

```
messages = []

if skipped_images_str:
    if imported_images_str:
        messages.append("Image(s) {0} were skipped as they already exist in Cisco Catalyst Center.".format(skipped_images_str))
        messages.append("Images {0} have been imported successfully.".format(imported_images_str))
    else:
        messages.append("Image(s) {0} were skipped as they already exist in Cisco Catalyst Center. No new images were imported.".format(skipped_images_str))
elif imported_images_str:
    messages.append("Image(s) {0} have been imported successfully into Cisco Catalyst Center.".format(imported_images_str))
else:
    messages.append("No images were imported.")

self.msg = " ".join(messages)

```

Reply:
updated the code